python/packages/ai/teams/app.py

```diff
@@ -329,25 +346,27 @@ async def _on_turn(self, context: TurnContext):
                 and context.activity.type == ActivityTypes.message
                 and context.activity.text
             ):
-                await self._ai.chain(context, state, self._options.ai.prompt)
+                await self._ai.chain(context, cast(StateT, state), self._options.ai.prompt)
+
+            await state.save(self._options.storage)
```

Comment:
The state saving conditions seems to be different that JS/C# SDK. Do we need some discussions with Corina/Kavin on this?

Reply:
Thats a good callout, I'll write it down and we can discuss it.